### PR TITLE
Run macos packaging in GHA.

### LINKED ISSUE #2428: macOS builds are broken
**Environment information**
* Operating System: macOS 10.15.6
* Cutter version: 1.12.0
* Obtained from:
  - [ ] Built from source
  - [x] Downloaded release from Cutter website or GitHub 
  - [ ] Distribution repository
* File format:

**Describe the bug**

Crashes on startup:

```
Application Specific Information:
dyld: launch, loading dependent libraries
DYLD_FRAMEWORK_PATH=/Applications/Cutter.app/Contents/MacOS/../Frameworks
DYLD_LIBRARY_PATH=/Applications/Cutter.app/Contents/MacOS/../Frameworks

Dyld Error Message:
  Library not loaded: /usr/local/opt/gettext/lib/libintl.8.dylib
  Referenced from: /Applications/Cutter.app/Contents/Frameworks/Python.framework/Versions/3.6/Python
  Reason: image not found
```

**To Reproduce**

Open Cutter.

**Expected behavior**

Continue to use IDA Pro :)

**Screenshots**

image

### PR DESCRIPTION
**Pull request by karliss**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

* Enable running of macOS packaging in GHA, most of the work done in dedbabde564ed688961cdc06828e39d6ca030e78
* Cleanup breakpad handling
* Have single version of main executable in folder expected by macOS and most tools instead of executable+symlink+shell script
* Handle Breakpad library lookup in more CMake way using FindBreakpad just like it's done on other platforms and packages
* Refactor error handling in some of the shell scripts to use `set -e` instead of `|| exit 1` for each command.
* Fix DMG background setup

This might need a followup PR making some small adjustments to make the devbuild process smoother. But first priority is making sure a release package can be built. 

**Test plan (required)**

* Download DMG artifact from GHA
* the package works on clean macOS installation without homebrew :heavy_check_mark: 
* sample python plugin works :heavy_check_mark: 
* decompiler works :heavy_check_mark: 

**Closing issues**

Closes https://github.com/rizinorg/cutter/issues/2525
Closes #2428

### CODE REVIEW

**Comment by karliss** [comment]
Breakpad framwork isn't properly included.

**Comment by XVilka** [comment]
Should be `dist/macos/DS_Store` removed? Seems committed mistakenly.

**Comment by karliss** [comment]
@XVilka Committed intentionally and named DS_Store instead of .DS_Store to prevent getting confused or accidentally overwritten by finder when opening folder. It is for describing background and icon placement in DMG. It isn't working quite right at the moment, need to double check I used correct image path. I might switch to apple script approach later if it turns out to work better.

File: dist/MacOSSetupBundle.cmake.in [hunk 1]
```diff
@@ -13,25 +17,33 @@ endmacro()
...
-# Move the executable from Resources/bin/Cutter to MacOS/Cutter
-# macdeployqt can't handle anything else
-file(RENAME "${CMAKE_INSTALL_PREFIX}/bin/Cutter" "${EXECUTABLE_DIR}/Cutter")
-# Make a symlink from the original location so CutterConfig.cmake won't be broken
-run_or_die(COMMAND ${CMAKE_COMMAND} -E create_symlink "../../MacOS/Cutter.bin" "${CMAKE_INSTALL_PREFIX}/bin/Cutter")
+set(FRAMEWORK_DIR "${BUNDLE_PATH}/Contents/Frameworks")
 
-# Copying the Info.plist will be done later again by CPack but we need it a bit earlier
-# so macdeployqt has enough info.
+## Copying the Info.plist will be done later again by CPack but we need it a bit earlier
```
Comment (karliss) [inline dist/MacOSSetupBundle.cmake.in:22]: Unnecessary change and double comment.

**Review by thestr4ng3r — APPROVED** [review]
Nice!
